File: hooks/KMS_EncryptionSettings/src/test/java/com/awscommunity/kms/encryptionsettings/AbstractTestBase.java
package com.awscommunity.kms.encryptionsettings;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;

import org.mockito.Mock;
import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
import software.amazon.cloudformation.proxy.HandlerErrorCode;
import software.amazon.cloudformation.proxy.Logger;
import software.amazon.cloudformation.proxy.OperationStatus;
import software.amazon.cloudformation.proxy.ProgressEvent;
import software.amazon.cloudformation.proxy.hook.targetmodel.HookTargetModel;

/**
 * Common unit tests logic base class.
 */
public abstract class AbstractTestBase {

    @Mock
    private AmazonWebServicesClientProxy proxy;

    @Mock
    private Logger logger;

    protected final AmazonWebServicesClientProxy getProxy() {
        return proxy;
    }

    protected final Logger getLogger() {
        return logger;
    }

    protected AbstractTestBase() {
        proxy = mock(AmazonWebServicesClientProxy.class);
        logger = mock(Logger.class);
    }

    protected final void testHandlerInternalFailure(final ProgressEvent<HookTargetModel, CallbackContext> response,
            final String message) {
        assertThat(response).isNotNull();
        assertThat(response.getMessage()).isEqualTo(message);
        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.HandlerInternalFailure);
        assertThat(response.getCallbackContext()).isNull();
        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
    }

    protected final void testInvalidRequest(final ProgressEvent<HookTargetModel, CallbackContext> response,
            final String message) {
        assertThat(response).isNotNull();
        assertThat(response.getMessage()).isEqualTo(message);
        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.InvalidRequest);
        assertThat(response.getCallbackContext()).isNull();
        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
    }

    protected final void testNonCompliant(final ProgressEvent<HookTargetModel, CallbackContext> response,
            final String message) {
        assertThat(response).isNotNull();
        assertThat(response.getMessage()).isEqualTo(message);
        assertThat(response.getStatus()).isEqualTo(OperationStatus.FAILED);
        assertThat(response.getErrorCode()).isEqualTo(HandlerErrorCode.NonCompliant);
        assertThat(response.getCallbackContext()).isNull();
        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
    }

    protected final void testCompliant(final ProgressEvent<HookTargetModel, CallbackContext> response,
            final String message) {
        assertThat(response).isNotNull();
        assertThat(response.getMessage()).isEqualTo(message);
        assertThat(response.getStatus()).isEqualTo(OperationStatus.SUCCESS);
        assertThat(response.getErrorCode()).isNull();
        assertThat(response.getCallbackContext()).isNull();
        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
    }
}

File: hooks/S3_AccessControl/src/test/kotlin/com/awscommunity/s3/accesscontrol/AbstractTestBase.kt
package com.awscommunity.s3.accesscontrol

import io.mockk.impl.annotations.MockK
import org.assertj.core.api.Assertions.assertThat
import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy
import software.amazon.cloudformation.proxy.HandlerErrorCode
import software.amazon.cloudformation.proxy.OperationStatus
import software.amazon.cloudformation.proxy.ProgressEvent
import software.amazon.cloudformation.proxy.hook.HookContext
import software.amazon.cloudformation.proxy.hook.HookHandlerRequest
import software.amazon.cloudformation.proxy.hook.targetmodel.HookTargetModel

abstract class AbstractTestBase {

    @MockK
    protected lateinit var proxy: AmazonWebServicesClientProxy

    protected fun getMockTargetModelWithoutAccessControl(): MutableMap<String, Any> {
        val targetModel: MutableMap<String, Any> = HashMap()
        val resourceProperties: MutableMap<String, Any> = HashMap()
        targetModel["resourceProperties"] = resourceProperties
        return targetModel
    }

    protected fun getMockTargetModelWithAccessControlSetToPrivate(): MutableMap<String, Any> {
        val targetModel: MutableMap<String, Any> = HashMap()
        val resourceProperties: MutableMap<String, Any> = HashMap()
        resourceProperties["AccessControl"] = "Private"
        targetModel["resourceProperties"] = resourceProperties
        return targetModel
    }

    protected fun getMockTargetModelWithAccessControlSetToPublicRead(): MutableMap<String, Any> {
        val targetModel: MutableMap<String, Any> = HashMap()
        val resourceProperties: MutableMap<String, Any> = HashMap()
        resourceProperties["AccessControl"] = "PublicRead"
        targetModel["resourceProperties"] = resourceProperties
        return targetModel
    }

    protected fun getMockHookHandlerRequest(targetName: String, targetLogicalId: String, targetModel: MutableMap<String, Any>): HookHandlerRequest {
        return HookHandlerRequest.builder().hookContext(
            HookContext.builder().targetName(targetName)
                .targetLogicalId(targetLogicalId).targetModel(HookTargetModel.of(targetModel)).build()
        )
            .build()
    }

    protected fun testNonCompliant(
        response: ProgressEvent<HookTargetModel, CallbackContext?>,
        message: String
    ) {
        assertThat(response).isNotNull()
        assertThat(response.message).isEqualTo(message)
        assertThat(response.status).isEqualTo(OperationStatus.FAILED)
        assertThat(response.errorCode).isEqualTo(HandlerErrorCode.NonCompliant)
        assertThat(response.callbackContext).isNull()
        assertThat(response.callbackDelaySeconds).isEqualTo(0)
    }

    protected fun testCompliant(
        response: ProgressEvent<HookTargetModel, CallbackContext?>,
        message: String
    ) {
        assertThat(response).isNotNull()
        assertThat(response.message).isEqualTo(message)
        assertThat(response.status).isEqualTo(OperationStatus.SUCCESS)
        assertThat(response.errorCode).isNull()
        assertThat(response.callbackContext).isNull()
        assertThat(response.callbackDelaySeconds).isEqualTo(0)
    }
}
